### ISSUE TITLE
arm/coproc: remove coproc from vcoproc_init parameter's list

### ISSUE THREAD

**Issue by andr2000** [xen-troops/xen#23]
Currently it is defined as:
```
    int (*vcoproc_init)(struct domain *, struct coproc_device *,
                        struct vcoproc_instance *);

```
But, vcoproc_instance already has coproc_device.

**Comment by aanisov** [comment]
It was my misunderstanding of the code. You changed the interface of the function, so that it returns the status and vcoproc is another return value of the function. But this seems to be a functional change and doubtful, as for me.

**Comment by andr2000** [comment]
because now vcoproc instance is being allocated and set up outside

**Comment by aanisov** [comment]
OK, I've got the point, but please remove as well `d` from function parameters.

**Comment by andr2000** [comment]
Fixed by: commit 742b978ab2b5ee9ed935ce83e5d9a267ea0a6659
